src/policy/fees.cpp

```diff
@@ -902,6 +902,12 @@ CFeeRate CBlockPolicyEstimator::estimateSmartFee(int confTarget, FeeCalculation
      * horizons so we already have monotonically increasing estimates and
      * the purpose of conservative estimates is not to let short term
      * fluctuations lower our estimates by too much.
+     *
+     * Note: Monotonically increasing estimates are not guaranteed in certain
+     * data-sparse scenarios. In particular, if estimateCombinedFee returns no
+     * valid data (-1) for some estimates for a lower target, but does return
+     * valid data for a higher target, the estimate can theoretically return a
+     * higher value for higher targets.
```

Comment:
```suggestion
     * Note: In certain rare edge cases, monotonically increasing estimates may not be 
     * guaranteed. Specifically, given two targets N and M, where M > N, if a sub-estimate
     * for target N fail to return a valid fee rate, while target M has valid fee rate for that
     * sub-estimate, target M may result in a higher fee rate 
     * estimate than target N.
     *
     * See: https://github.com/bitcoin/bitcoin/issues/11800#issuecomment-349697807
 ```
I think the suggestion describes scenario better as stated in https://github.com/bitcoin/bitcoin/issues/11800#issuecomment-349697807.

This will be even better when we specify which sub-estimate failures break the monotonic
behavior of the fee rate estimates. The comments mention that we check the maximum 
confirmations for shorter time horizons to preserve this behavior. 

This applies to the first three sub-estimates for economical estimates and the first two for conservative estimates.